signer/local/local.go

```diff
@@ -29,14 +32,21 @@ import (
 	"github.com/google/certificate-transparency-go"
 	"github.com/google/certificate-transparency-go/client"
 	"github.com/google/certificate-transparency-go/jsonclient"
+
+	zx509 "github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint"
+	"github.com/zmap/zlint/lints"
 	"golang.org/x/net/context"
 )
 
 // Signer contains a signer that uses the standard library to
 // support both ECDSA and RSA CA keys.
 type Signer struct {
-	ca         *x509.Certificate
-	priv       crypto.Signer
+	ca   *x509.Certificate
+	priv crypto.Signer
+	// lintPriv is generated randomly when pre-issuance linting is configured and
+	// used to sign TBSCertificates for linting.
+	lintPriv   crypto.Signer
```

Comment:
Since this is an internal implementation detail, we have the luxury of explicitly typing it (`*ecdsa.PrivateKey`). That saves a `switch` later on, and also makes it a little clearer what the field does when reading this part of the code.

Reply:
I thought it was slightly cleaner to use an interface/type switch to make it easier to use different key types in the future. It's _probably_ overkill, I'd be open to changing it to the explicit type if folks feel strongly.